optimization of SiStip RecHits Validation

ngrenz opened the pull request:
Optimization by deleting of unused variables, set of variables only one time and restructure of if-statements.

In addition some suggestion from the pull request #10586 (with same name and 
content as this pull request) were implemented.
The suggestion, with major reconstruction of the code like rewritting the code for the use of a enum will be made in a later pull request.

The outputs are supposed to be unchanged.
Any comments and suggestions are welcome.

slava77devel committed:
add manual bunchSpacing setting; set run 253000 as a 25 ns switchover

slava77devel committed:
set run 253000 as a 25 ns switchover; actually use the bunchSpacingManual

slava77devel committed:
in customiseDataRun2Common: add Stage1 emulator and DQM customs; added 25ns and 50ns specific customiseDataRun2Common_25ns and customiseDataRun2Common_50ns; use customiseDataRun2Common_25ns in prompt and express settings; add _50ns versions of Prompt and Express to override things that don't get picked up automatically

Conflicts:
	Configuration/DataProcessing/python/RecoTLR.py
... picking mine

slava77devel committed:
ppRun2at50ns scenario ... hopefully for a very limited use (not adding anything for 0Tesla or hcal nzs)

Conflicts:
	Configuration/DataProcessing/test/run_CfgTest.sh
... picked mine

slava77devel committed:
more explicit about use case of 50ns; actually add customisePromptRun2_50ns

slava77devel committed:
add bunchSpacing to pset validation

mulhearn committed:
add l1reco hook for stage1

slava77devel committed:
add bunchSpacing to pset validation as a full parameter

slava77devel committed:
switched to customiseL1RecoForStage1

slava77devel committed:
move stage1 only to methods to be used in 2015C data; cleaned up import statements which begin to give warnings in 75X/76X

slava77devel committed:
add hasattr to customized parameters in customiseDataRun2Common

matteosan1 committed:
first implementation

matteosan1 committed:
final update

matteosan1 committed:
fix memory leak and tag label

cmsbuild commented:
A new Pull Request was created by @ngrenz  for CMSSW_7_6_X.

optimization of SiStip RecHits Validation

It involves the following packages:

Validation/TrackerRecHits


@cmsbuild, @danduggan, @deguio can you please review it and eventually sign? Thanks.
@VinInn, @threus, @wmtford this is something you requested to watch as well.
You can sign-off by replying to this message having '+1' in the first line of your reply.
You can reject by replying  to this message having '-1' in the first line of your reply.
If you are a L2 or a release manager you can ask for tests by saying 'please test' in the first line of a comment.
@degano you are the release manager for this.
You can merge this pull request by typing 'merge' in the first line of your comment.

ngrenz commented:
@boudoul FYI

makortel commented on Validation/TrackerRecHits/src/SiStripRecHitsValid.cc:
```diff
@@ -417,17 +408,17 @@ void SiStripRecHitsValid::rechitanalysis(SiStripRecHit2D const rechit,const Stri
       if(dist<mindist){
 	mindist = dist;
-	closest = (*m);
+	closest = m;
```
You could also avoid copying PSimHit here, e.g. by taking (const) pointer.

makortel commented on Validation/TrackerRecHits/src/SiStripRecHitsValid.cc:
```diff
@@ -496,22 +487,22 @@ void SiStripRecHitsValid::rechitanalysis_matched(SiStripMatchedRecHit2D const re
 	  mindist = dist;
 	  closestPair = hitPair;
-	  closest = (*m);
+	  closest = m;
```
Same here, no need to copy.

makortel commented on Validation/TrackerRecHits/src/SiStripRecHitsValid.cc:
```diff
@@ -594,26 +584,31 @@ void SiStripRecHitsValid::createMEs(DQMStore::IBooker & ibooker,const edm::Event
-      if (det_layer_pair.first == "TIB") {
-	substructure.getTIBDetectors(activeDets,layerDetIds,lnumber,0,0,0);
-      } else if (det_layer_pair.first == "TOB") {
+      std::string lname = det_layer_pair.first; 
```
const std::string&

makortel commented on Validation/TrackerRecHits/src/SiStripRecHitsValid.cc:
```diff
@@ -594,26 +584,31 @@ void SiStripRecHitsValid::createMEs(DQMStore::IBooker & ibooker,const edm::Event
-      } else if (det_layer_pair.first == "TOB") {
+      std::string lname = det_layer_pair.first; 
+      std::vector<uint32_t> layerDetIds;
+      std::string tec = "TEC", tid = "TID", tob = "TOB", tib = "TIB";        
```
constexpr char tec[] = "TEC" etc?

VinInn replied:
no, this I asked for!
so that the comparison is string against string, not string against const char *

Dr15Jones replied:
Still, all these strings should be ```const```.

makortel replied:
But this way the strings are created explicitly.

So the compilers still generate temporary strings even if standard requires `operator==(char *, string)` and `operator==(string, char*)`? Anyway, then I'd say the "problem" is the use of operator==, not char*, so `lname.compare(tec) == 0` could be the way to go then.

VinInn replied:
maybe we have to check that with gcc49x and c++14. The point is that we are still using old string implementation due to clang compatibility

gpetruc committed:
DataFormats/FWLite: avoid crash in MultiChainEvent constructor if the primary input files have all zero events

gpetruc committed:
DataFormats/FWLite: ChainEvent: fix also toBegin and atEnd, so that 'for event in events' still works (and does nothing) when there are no files

deguio commented:
-1
I guess that this one is still waiting for a commit to address the comments.
thanks to all,
F.

pietverwilligen committed:
pass by const reference

Commit:
fix pfClusterMET test

pietverwilligen committed:
pass by const reference

pietverwilligen committed:
pass by const reference

pietverwilligen committed:
pass by const reference

pietverwilligen committed:
pass by const reference

pietverwilligen committed:
add protection against dividing by zero

pietverwilligen committed:
99999 --> running_max

cmsbuild committed:
Merge pull request #10534 from cms-tsg-storm/otherUpdatesToThe25nsHltMenu_75X

HLT update for 25ns (75X)

pietverwilligen committed:
add functionality to ask number of layers

pietverwilligen committed:
removed magic number 5 = number of layers in ME0 chamber - 1

pietverwilligen committed:
removed magic number 5 = number of layers in ME0 chamber - 1

pietverwilligen committed:
use use reco::deltaPhi

cmsbuild committed:
Merge CMSSW_7_5_X into CMSSW_7_6_X.

covarell committed:
add lhe generic filter

Commit:
fix 76X pfClusterMEt test

smorovic committed:
detection of "stream" (not case sensitive) sequence in stream suffix which will result in exception.

smorovic committed:
detection of "stream" (not case sensitive) sequence in stream suffix which will result in exception.

apfeiffer1 committed:
example of a payload-to-XML C++ plugin which can/will be called from the cond2xml.py python class

pietverwilligen committed:
precompute layer,phi,eta in separate loops

ginnocen committed:
new methods for D filters

dgulhan committed:
Update hiCaloJetsForTrk_cff.py

dgulhan committed:
Update HIInitialJetCoreClusterSplitting_cff.py

dgulhan committed:
Update hiJetCoreRegionalStep_cff.py

dgulhan committed:
Update HIInitialJetCoreClusterSplitting_cff.py

ginnocen committed:
new methods for D filters

apfeiffer1 committed:
if called w/o arguments, assume it is run as a unit-test and simply do nothing

davidlange6 committed:
Merge pull request #10694 from mark-grimes/fixPhase1TkCustoms76X

Fix phase1TkCustoms for recent changes to defaults

pietverwilligen committed:
use std::unique_ptr for instance of ME0SegFit class

pietverwilligen committed:
use std::unique_ptr for instance of ME0SegFit class

pietverwilligen committed:
use std::unique_ptr for instance of ME0SegFit class

cmsbuild committed:
Merge pull request #10792 from dmitrijus/viz_online_fix_pr_75x

Fix online DQM configuration for visualization (75x)

cmsbuild committed:
Merge pull request #10684 from prbbing/BackportFrom76X_10681_For75X

Backport from 76X 10681, added new HT parking paths and changed the b…

cmsbuild committed:
Merge pull request #10531 from cms-tsg-storm/otherUpdatesToThe25nsHltMenu_76X

HLT update for 25ns (76X)

cmsbuild committed:
Merge pull request #10296 from kfjack/Update_MuTauSkim_w_NewTauIso

Update MuTauSkim_cff with new Tau POG Isolation from Ram

nhanvtran committed:
reindenting

cmsbuild committed:
Merge pull request #10666 from cmkuo/mydev_75x

add low pt photons (10-14 GeV) to miniAOD for 75x

cmsbuild committed:
Merge pull request #10657 from echapon/75X_highlevelHiMuonReco

Run "high level muon reconstruction" for heavy ions

cmsbuild committed:
Merge pull request #10595 from alja/dima-750-cfg

75x Fireworks:Update collection entry in configuration files

pietverwilligen committed:
pass arguments as const for function isGoodToMerge

pietverwilligen committed:
pass arguments as const for function isGoodToMerge

cmsbuild committed:
Merge CMSSW_7_5_X into CMSSW_7_6_X.

pietverwilligen committed:
define class constant running_max as static constexpr float

capalmer85 committed:
updated scripts

capalmer85 committed:
remove unnecessary code; changed global tag

pietverwilligen committed:
moved calculation of time to the segment algorithm

pietverwilligen committed:
moved calculation of time to the segment algorithm

pietverwilligen committed:
moved calculation of time to the segment algorithm

pietverwilligen committed:
moved calculation of time to the segment algorithm

pietverwilligen committed:
minor modification of LogDebug msg

pietverwilligen committed:
adding original version in 760pre2

pietverwilligen committed:
PR version now in 760pre2

pietverwilligen committed:
PR version now in 760pre2

pietverwilligen committed:
changed class version according to compilation error msg

pietverwilligen committed:
PR version now in 760pre2

pietverwilligen committed:
PR version now in 760pre2

pietverwilligen committed:
PR version now in 760pre2

pietverwilligen committed:
PR version now in 760pre2

pietverwilligen committed:
PR version now in 760pre2

pietverwilligen committed:
PR version now in 760pre2

pietverwilligen committed:
PR version now in 760pre2

pietverwilligen committed:
PR version now in 760pre2

pietverwilligen committed:
PR version now in 760pre2

pietverwilligen committed:
PR version now in 760pre2

pietverwilligen committed:
PR version now in 760pre2

pietverwilligen committed:
PR version now in 760pre2

pietverwilligen committed:
PR version now in 760pre2

pietverwilligen committed:
PR version now in 760pre2

pietverwilligen committed:
PR version now in 760pre2

pietverwilligen committed:
PR version now in 760pre2

pietverwilligen committed:
PR version now in 760pre2

pietverwilligen committed:
protect against division by zero

fwyzard committed:
hltGetConfiguration: switch run:NNNNNN to use ConfDB v2

fwyzard committed:
hltGetConfiguration: switch run:NNNNNN to use ConfDB v2

pietverwilligen committed:
hide stringstreams necessary for formatting output behind EDM_ML_DEBUG

pietverwilligen committed:
cleaned up comments

pietverwilligen committed:
keep valuable LogTrace msg

pietverwilligen committed:
removed comments starting with $

pietverwilligen committed:
removed comments starting with $

pietverwilligen committed:
Merge pull request #16 from cms-sw/CMSSW_7_6_X

PR_ToBringUpToDateMyBranch_ME0Segment_PR_v3

pietverwilligen committed:
removed comments starting with $

slava77devel committed:
add dqmSeq option to the recoinator

slava77devel committed:
add dqmSeq option to the recoinator

nhanvtran committed:
a few line-by-liners and two mis-initializations

lgray committed:
address most of code review comments (some are left for another PR)

lgray committed:
fix issues spotted by Jean-Baptiste

cmsbuild committed:
Merge pull request #10828 from fwyzard/use_ConfDBv2_for_run_number_76x

hltGetConfiguration: switch run:NNNNNN to use ConfDB v2

cmsbuild committed:
Merge pull request #10808 from smorovic/empty-LS-mode-76X

metadata JSON files produced for empty lumisections in DAQ (76X)

cmsbuild committed:
Merge pull request #10706 from alberto-sanchez/evtgen

Fixing horrible typo. Also add a Log message for force decay and clea…

davidlange6 committed:
Merge pull request #10609 from schoef/CMSSW_7_6_X_Devel2

MET Filter update (port of #10604)

cmsbuild committed:
Merge pull request #10822 from ginnocen/mybranchDfilters

new methods for D filters

cmsbuild committed:
Merge pull request #10819 from apfeiffer1/toxml-plugin

Example of a payload-to-XML C++ plugin

dgulhan committed:
Update hiCaloJetsForTrk_cff.py

dgulhan committed:
Update HIInitialJetCoreClusterSplitting_cff.py

davidlange6 committed:
Merge pull request #10258 from duanders/scoutingpackingcode

Data scouting classes and producers (76X)

dgulhan committed:
Update hiJetCoreRegionalStep_cff.py

covarell committed:
consums migration

yetkinyilmaz committed:
option added for das query

pietverwilligen committed:
Merge pull request #17 from cms-sw/CMSSW_7_6_X

PR_ToBringUpToDateMyBranch_ME0Segment_PR_v4

jpata committed:
reverted pat changes

pietverwilligen committed:
Merged GEMSegment_76X_v2 from repository pietverwilligen

Commit:
condformats_serialization_generate.py: fix include paths and fail on serious issues

- The script itself uses libClang for parsing C++ source files, thus we
must use include files as seen by Clang. Note, this still points to GNU
C++ standard library in GCC package, but there are special headers which
are compiler specific, those are picked up from Clang package.

- Print severity level names instead numbers.

- Fail if there are 'Error' or/and 'Fatal' level diagnotic messages.

- Remove obsolete ignore rules. They should not be triggered as long as
  include paths and arguments passed to libClang are valid.

- This should resolve issue where `get_statement` could return `None`,
  because compiler couldn't do a correct job due to missing/wrong
include paths or/and arguments.

- Required for Clang 3.7.0.

Signed-off-by: David Abdurachmanov <David.Abdurachmanov@cern.ch>

pietverwilligen committed:
delete outdated test cfg file

pietverwilligen committed:
delete outdated test cfg file

pietverwilligen committed:
delete outdated test cfg file

pietverwilligen committed:
delete outdated test cfg file

yetkinyilmaz committed:
option added for das query

pietverwilligen committed:
migrate to consumes

pietverwilligen committed:
migrate to consumes

pietverwilligen committed:
migrate to consumes

ngrenz committed:
string

pietverwilligen committed:
migrate to consumes

pietverwilligen committed:
remove empty methods and call ES for each event

pietverwilligen committed:
use magneticfield_cff

jpata committed:
added back new import path

boudoul committed:
Merged refs/pull/10804/head from repository cms-sw

boudoul committed:
relval updates  on top of 760pre3

davidlange6 committed:
Merge pull request #10816 from covarell/76X_genfilter

add lhe generic filter

davidlange6 committed:
Merge pull request #10829 from slava77/CMSSW_7_6_X_2015-08-17-1100/t0DqmSeq

add dqmSeq option to the recoinator

davidlange6 committed:
Merge pull request #10471 from apfeiffer1/conddb-copy-gts

Update conddb tool to make a copy of a full GT

davidlange6 committed:
Merge pull request #10830 from slava77/CMSSW_7_5_X_2015-08-17-1100/t0DqmSeq

add dqmSeq option to the recoinator (same as #10829)

ianna committed:
* Replace legacy analyzer with thread-friendly

yetkinyilmaz committed:
avoid filling the linux buffer by doing the stdout.read() first

cmsbuild committed:
Merge pull request #10827 from fwyzard/use_ConfDBv2_for_run_number_75x

hltGetConfiguration: switch run:NNNNNN to use ConfDB v2

lgray committed:
actually assign the calculated weight

ngrenz committed:
i removed

ggovi committed:
Improved handling of customized GT via toGet: set default snapshotTime to infinity

pietverwilligen committed:
conclude merge

pietverwilligen committed:
Merge pull request #18 from cms-sw/CMSSW_7_6_X

PR_ToBringUpToDateMyBranch_ME0Segment_PR_v5

lgray committed:
remove ESPrefers for GBRD conditions

gpetruc committed:
Add unit tests of multichain with empty files

pietverwilligen committed:
update testME0DetId: max EtaPartition for now is 1

pietverwilligen committed:
Merge branch 'GEMSegment_76X_v1' of https://github.com/pietverwilligen/cmssw into GEMSegment_76X_v1

davidlange6 committed:
Merge pull request #10377 from duanders/scoutingpackingcode75X

Data scouting classes and producers (75X)

davidlange6 committed:
Merge pull request #10807 from smorovic/empty-LS-mode-75X

metadata JSON files produced for empty lumisections in DAQ (75X)

cmsbuild committed:
Merge pull request #10823 from ginnocen/mybranchDfilters75X

new methods for D filters -- backports of  #10822

cmsbuild committed:
Merge pull request #10722 from prbbing/BackportFrom76X_10719_To75X

Added ne displacedJets paths

cmsbuild committed:
Merge pull request #10702 from richard-cms/EnableGEDPhotonsHI_75X

Put GED photons into HI event content and DQM pathways (backport 75X #10700)

cmsbuild committed:
Merge pull request #10696 from cms-l1t-offline/Stage1HFBitCountsAlgo_75X

 Add HF Minimum Bias algorithm to Stage1 Emulator (backport #10695)

cmsbuild committed:
Merge pull request #10685 from muell149/HLTObjectMonitoring_75x_v3

adding HLT object monitoring to HLT online DQM

cmsbuild committed:
Merge pull request #10632 from fioriNTU/PixBarrelEfficiency75

PixelBarrelEfficiency75

cmsbuild committed:
Merge pull request #10541 from rappoccio/rappoccio_B2GDQM_V297_75x

Adjusting trigger paths for B2G DQM : 75x

cmsbuild committed:
Merge CMSSW_7_5_X into CMSSW_7_6_X.

hengne committed:
add back wrongly removed workflow SMS-T1tttt_mGl-1500_mLSP-100_13 in commit 14bf693e6875

hengne committed:
Merge branch 'from-CMSSW_7_6_0_pre3_relval' into from-CMSSW_7_6_0_pre3_relval_rebase

hengne committed:
rebase -- Merge branch 'CMSSW_7_6_X' into from-CMSSW_7_6_0_pre3_relval_rebase

lgray committed:
more code review

ianna committed:
* Replace legacy analyzer with thread-friendly
* Clean up static analyzer warnings

Commit:
Update Scal (Online DQM) configuration to work with our new paths.

(cherry picked from commit 21dd6e744734e884ad54b108a39913942c3b6b4b)

Commit:
Move scal config to the correct folder.

(cherry picked from commit bb08d098ef97568fc22067cbe7809b9e878d06d0)

ianna committed:
* Merge files

archiron committed:
y axis modification of scl_EoEtrueVsrecOfflineVertices histo

Commit:
Adding CaloParams and CaloConfig types to CondDB

Commit:
Adding CaloParams and CaloConfig types to CondDB (repeat)

lgray committed:
make full5x5 or fraction-ified shower shapes configurable in regressions

ianna committed:
* One last EDAnalyzer fixed
* Thread-safety improved

nhanvtran committed:
speed up PATPackedCandidateProducer.cc

nhanvtran committed:
more speeding up

Dr15Jones committed:
Fix thread-safety issue with StreamerInputSource

The previous way StreamerInputSource handled reading back RefCores
lead to the probability that when dereferenced the RefCore would read
from the wrong EventPrincipal when the job was using multiple streams
becuase it was reusing the same EventPrincipalHolder for each event.
The code has now been changed to make a new EventPrincipalHolder for
each event.

Dr15Jones committed:
Fix thread-safety issue with StreamerInputSource

The previous way StreamerInputSource handled reading back RefCores
lead to the probability that when dereferenced the RefCore would read
from the wrong EventPrincipal when the job was using multiple streams
becuase it was reusing the same EventPrincipalHolder for each event.
The code has now been changed to make a new EventPrincipalHolder for
each event.

nhanvtran committed:
one more small optimization

Commit:
DQM/PixelLumi: definite static const int members

The patch resolves issue with Clang compiler (3.7.0).

N3690 (should be C++11 standard), 9.4.2/3

    ...
    The member shall still be defined in a namespace scope if it
    is odr-used (3.2) in the program and the namespace scope definition
    shall not contain an initializer.

9.4.2/4 talks about how `const static` data members are being handled.

Also standard says that no diagnostic is required by compiler.

Signed-off-by: David Abdurachmanov <David.Abdurachmanov@cern.ch>

nhanvtran committed:
diagnostics turned off to speed it up

Commit:
SiStripRecHitMatcher.h: remove non-POD VLA

Clang 3.7.0 will reject non-POD VLAs:

    RecoLocalTracker/SiStripRecHitConverter/interface/SiStripRecHitMatcher.h:188:19:
    error: variable length array of non-POD element type
    'matcherDetails::StereoInfo'

Unknown why this is not triggered in ~3.6.0.

Signed-off-by: David Abdurachmanov <David.Abdurachmanov@cern.ch>

bbockelm committed:
Fix sigstack printer.

We observed that, when CLONE_VFORK is used, gdb sometimes did not
print out the thread associated with the calling process.  As this
is the thread which caused the SEGV in the first place, it made the
sigstack printer less-than-useful.

We were not able to identify why it worked with the CLONE_VFORK in
some cases.

Further, we observed that instances exiting with SEGV/BUS/ILL always
returned exit status 134 (SIGABRT).  This rethrows the signal in
order to get the correct exit status.

(cherry picked from commit 0ad1c49696730150d9115c912de775e6653285e1)

jhgoh committed:
propagate only for valid track momentum

davidlange6 committed:
Merge pull request #10869 from jhgoh/fixCrashNullPlane75

Fix crash due to invalid track status 76X

cmsbuild committed:
Merge pull request #10856 from Dr15Jones/threadSafeStreamerInputSource_7_6

Fix thread-safety issue with StreamerInputSource

cmsbuild committed:
Merge pull request #10846 from ianna/geom-legacy-analyzer

Replace Legacy EDAnalyzer with Thread-friendly in Geometry

cmsbuild committed:
Merge pull request #10843 from ggovi/new-conddb-fix-essource-input-params-3

Improved handling of customized GT via toGet

cmsbuild committed:
Merge pull request #10825 from capalmer85/pcc-ntupler-76x-update-v1

Pcc ntupler 76x update v1

cmsbuild committed:
Merge pull request #10818 from mmarionncern/fixMETTest76Xv2

Fix of the 76X recoMET tests

davidlange6 committed:
Merge pull request #10601 from cschomak/CMSSW_7_6_0_pre2_APETool

APE tool implemented

davidlange6 committed:
Merge pull request #10868 from jhgoh/fixCrashNullPlane75

Fix crash due to invalid track status 75X

cmsbuild committed:
Merge pull request #10848 from dmitrijus/scal_config_fix_75x

Update Scal (Online DQM) configuration to work with our new setup (76x)

davidlange6 committed:
Merge pull request #10865 from slava77/CMSSW_7_5_X_2015-08-18-2300/run2at25ns

forward port of dataProcessing changes for 25ns

davidlange6 committed:
Merge pull request #10789 from threus/updateOccupancyScript_75x

tiny update of occupancy script

cmsbuild committed:
Merge pull request #10857 from Dr15Jones/threadSafeStreamerInputSource_7_5

Fix thread-safety issue with StreamerInputSource

cmsbuild committed:
Merge pull request #10849 from dmitrijus/scal_config_fix_75x

Update Scal (Online DQM) configuration to work with our new setup (75x)

cmsbuild committed:
Merge pull request #10765 from mmarionncern/fixMETTest75X

Update the default relval for the recoMET tests

cmsbuild committed:
Merge pull request #10749 from lgray/move_slimmedPSU_to_MCCustomize_75X

Move slimmedAddPileupInfo to miniAOD_customizeMC (75X)

cmsbuild committed:
Merge pull request #10300 from kfjack/Update_MuTauSkim_w_NewTauIso_7_5_X

Update MuTauSkim_cff with new Tau POG Isolation from Ram (7_5_X)

cmsbuild committed:
Merge CMSSW_7_5_X into CMSSW_7_6_X.

apfeiffer1 committed:
add two more older releases. 725 is OK, 7.1.19 can not read more recent ones, as expected

apfeiffer1 committed:
do not take into accound known failures in 7.1.x cycle

mmusich committed:
Adding e/gamma regression from Ecal Multifit

davidlange6 committed:
Merge pull request #10783 from ahinzmann/adjustminiaod170

Adjust ak8 jet threshold in MiniAOD

davidlange6 committed:
Merge pull request #10767 from apfeiffer1/condRegressionTest

First version of a "unit-test" like regression test for conditions

davidlange6 committed:
Merge pull request #10625 from ahinzmann/PUjetIDcleanup76

Cleanup duplicate pileup jet ID configurations

pietverwilligen committed:
abs value of reco::deltaPhi since it returns values in [-pi,pi]

cmsbuild committed:
Merge pull request #10840 from boudoul/from760pre3relval

From760pre3relval

hengne committed:
rebase again - Merge branch 'CMSSW_7_6_X' into from-CMSSW_7_6_0_pre3_relval_rebase

MarkBaber committed:
Addition of AlphaT DQM monitoring

davidlange6 committed:
Merge pull request #10885 from bbockelm/fix-sigstack-75x

Fix sigstack printer.

davidlange6 committed:
Merge pull request #10777 from dgulhan/CS_75X

Pixel cluster splitting and jet core iteration for HI Tracking

davidlange6 committed:
Merge pull request #10754 from ferencek/MoveGBRForestWriter_from-CMSSW_7_5_1

Moving GBRForestWriter to CommonTools/Utils: 75X

davidlange6 committed:
Merge pull request #10597 from nhanvtran/puppi-75x-miniaodpass2-v2

update PUPPI based on DP note

cmsbuild committed:
Merge pull request #10872 from mmusich/GTs_for_75_v2

Adding e/gamma regression from Ecal Multifit in Run1/Run2 GTs

cmsbuild committed:
Merge CMSSW_7_5_X into CMSSW_7_6_X.

wmtan committed:
Move parentage code out of the base class

davidlange6 committed:
Merge pull request #10889 from wmtan/OutputModuleThreadSafety1

Move parentage code out of the OutputModule base class

davidlange6 committed:
Merge pull request #10197 from jpata/combmva_softel_tmvaeval_from_7_6_X

Combined MVA computer based on TMVA (CombinedMVAV2)

davidlange6 committed:
Merge pull request #10172 from mark-grimes/eraUpdatesStandardSequences

Update Run 2 eras to match customisation functions

davidlange6 committed:
Merge pull request #10187 from mark-grimes/eraUpdatesFastSim

Updates the Run2 era to match recent changes in the customisations

cmsbuild committed:
Merge pull request #10859 from davidlt/PixelLumiDQM-lastBunchCrossing

DQM/PixelLumi: definite static const int members

cmsbuild committed:
Merge pull request #10708 from makortel/trkThreaded

Migrate bunch of tracking related modules to one/stream/global

Commit:
fix trigger feeding the pixel lumi dqm to match the VdM menu

mbandrews committed:
forcing EE qT to be always true

davidlange6 committed:
Merge pull request #10897 from gpetruc/FWLite-MultiChainEvent-fix-0ev-crash-76X

FWLite MultiChainEvent: avoid crash on empty files (76X)

andrewj314 committed:
Bugfix adding absolute value to deltaPhi, changing BOOSTFOREACH loops to auto iterators

davidlange6 committed:
try fixing consumes, then turn off unit test

davidlange6 committed:
Merge pull request #10900 from mojoe137/pixelLumiDqmFixForVdM-76X

fix trigger feeding the pixel lumi dqm to match the VdM menu

davidlange6 committed:
Merge pull request #10860 from davidlt/SiStripRecHitMatcher-cache

SiStripRecHitMatcher.h: remove non-POD VLA

davidlange6 committed:
Merge pull request #10908 from davidlange6/turnoffunittest

physicstools unit test removed

davidlange6 committed:
Merge pull request #10850 from kkotov/addingCaloParamsConfig2DB

Adding CaloParams and CaloConfig types to CondDB

davidlange6 committed:
Merge pull request #10689 from lgray/regressionModifier_for76X

Modifier to apply energy corrections to photons and electrons (miniAOD) for 76X

davidlange6 committed:
Merge pull request #10838 from yetkinyilmaz/das_query_76X_03

Option for DAS query in ConfigBuilder 76X

Martin-Grunewald committed:
Fix DQM py files for ConfDB parsing

lgray committed:
change photon regressions back to fractionized shower shapes (confirmed by Benoit)

lgray committed:
revert everything to fractionized shower shapes for regressions

cmsbuild committed:
Merge pull request #10837 from davidlt/fix-boost-serial

condformats_serialization_generate.py: fix include paths and fail on serious issues

cmsbuild committed:
Merge pull request #10811 from KiSooLee/cms76X_HLTDQM

HLT BPTX path for HI added

cmsbuild committed:
Merge pull request #10930 from cmkuo/mydev_76x_ESDQM

fix a bug for the number of good ES rechits

cmsbuild committed:
Merge pull request #10921 from lgray/switch_photon_regression_to_fractions

Change photon & electron regressions back to fractionized shower shapes

cmsbuild committed:
Merge pull request #10871 from archiron/EoEtrue_Vertices_Histos_modif

Electron Validation : modification of scl_EoEtrueVsrecOfflineVertices histo

cmsbuild committed:
Merge pull request #10805 from cms-tau-pog/DynamicStripReco_v4

Dynamic strip reco v4

cmsbuild committed:
Merge pull request #10905 from mbandrews/ecalEEAllGood76x

forcing EE qT to be always true

cmsbuild committed:
Merge pull request #10874 from hengne/from-CMSSW_7_6_0_pre3_relval_rebase

Add back a workflow removed by accident in commit 14bf693

davidlange6 committed:
Merge pull request #10775 from dgulhan/CS_76X

Pixel cluster splitting and jet core iteration for HI Tracking

davidlange6 committed:
Merge pull request #10485 from pietverwilligen/GEMSegment_76X_v1

Gem segment 76

davidlange6 committed:
Merge pull request #10914 from Martin-Grunewald/FixForConfDBparsing

Fix DQM py files for ConfDB parsing (76X)

davidlange6 committed:
Merge pull request #10877 from MarkBaber/CMSSW_7_6_X_DQM_AlphaT_Update_2

CMSSW_7_6_X DQM updates for AlphaT HLT paths (v2)

ngrenz committed:
Merged ngrenz/ngrenz_SiStip_RecHitsValidation from repository ngrenz

ngrenz committed:
pointer

ngrenz commented:
sorry, with my last commit added changes from other people.

It is easier to make a new pull-request than try to fix this, therefore I will close this request.

Thanks for the help and sorry for the inconvenience.
